### DISCUSSION TITLE
Cadence Based Motor Speed Control

### DISCUSSION THREAD

**Discussion by EigenMania** [danielnilsson9/bbs-fw#122] [Ideas]
I've been experimenting with a new control mode for my 52V. BBSHD, and I am curious to get some feedback from the community. 

The basic idea is to set the speed of the motor to match (or only slightly exceed) the pedalling cadence. This solves some performance characteristics of the current firmware that I do not find desirable (given my pedalling riding style). 

1. If you pedal faster than the motor speed setpoint for a particular assist level and gear ratio, then you get no extra support from the motor (feels like pedalling into some "resistance").

2. If you pedal up to speed and then slow your cadence down, but continue pedalling lightly, the bike will continue to try and reach the programmed target speed / cadence (possibility for ghost / clown pedalling). 

3. Low cadence control of the bike is nearly impossible in the higher support levels, since as soon as you start pedalling, the bike is whisked away as the current ramps up.

So far my idea seems to have worked :) Would be cool to get some test feedback from this group, especially if someone shares these problems with me. 

Here you can download the .HEX (for BBSHD), the config tool, and a sample config XML (which you can load and then modify as necessary): https://www.dropbox.com/s/h5sc2pcuqp7r085/Cadence_Based_Motor_Speed_Control.zip?dl=0

You can also see the branch I am working on here: https://github.com/EigenMania/bbs-fw/tree/feature/variable_target_cadence

Looking forward to hearing some feedback!

[Cadence_Based_Motor_Speed_Control.zip](https://github.com/danielnilsson9/bbs-fw/files/11417976/Cadence_Based_Motor_Speed_Control.zip)

**Comment by silenec** [comment]
Sounds nice, I would like to test it. Do you think this could be made to work with BBS02 too if the BBS02 instability issues get solved?

**Reply by CitizenDev** [reply]
I wonder the same if BBS02 controller RAM is limiting this feature somehow

**Reply by EigenMania** [reply]
Yes, RAM is a problem. BBS02 is completely out of space, but once this is addressed I can also compile for it.

**Reply by kb79000** [reply]
> Oui, la RAM est un problème. BBS02 manque complètement d'espace, mais une fois que cela est résolu, je peux également compiler pour cela.

Hello BBS02 and BBS02B have the same amount of RAM? Thank

**Comment by kklmm** [comment]
Hi @EigenMania, can you compile the latest version of this branch? The hex and the gui please.
I desperately want to test this functionality as I live in the alps and terrain is very uneven making pedals useless with stock PAS functionality.  Forced to use throttle for controllable power output.

Or please provide instructions how to compile it ourselves. Thank you. Great work.
